api/src/utils/generics/lifecycle-hook-manager.ts

```diff
@@ -57,6 +63,7 @@ export class LifecycleHookManager {
   }
 
   public static attach(model: ModelDefinition): ModelDefinition {
+    this.models.push(model);
```

Comment:
let use  Set() and throw an error if we try to attach the same model twice.

Reply:
Sure